ballerina/client.bal

```diff
@@ -357,760 +358,425 @@ public isolated client class Client {
-        string path = utils:prepareQueryUrl([API_BASE_PATH, QUERY], [Q], [receivedQuery]);
-        json res = check self.get(path);
-        return toSoqlResult(res);
+    # + bulkJobId - Id of the bulk job
+    # + bulkOperation - The processing operation for the job
+    # + return - `()` if successful else `error`
+    isolated remote function deleteJob(string bulkJobId, BulkOperation bulkOperation) returns error? {
+        string path = utils:prepareUrl([API_BASE_PATH, JOBS, bulkOperation, bulkJobId]);
+        http:Response response = check self.salesforceClient->delete(path);
+        if response.statusCode != 204 {
+            return error("Error occurred while deleting the bulk job. " + 
+                (check response.getJsonPayload()).toString());
+        }
     }
 
-    # If the query results are too large, retrieve the next batch of results using the nextRecordUrl.
-    #
-    # + nextRecordsUrl - URL to get the next query results
-    # + return - `SoqlResult` record if successful. Else, the occurred `Error`.
-    #
-    # # Deprecated
-    # This function is deprecated as it is related to pagination of results in `query(string soql)`
-    # Use the new and improved `query(string soql)` function instead.
-    @display {label: "Get Next Query Result"}
-    @deprecated
-    isolated remote function getNextQueryResult(@display {label: "Next Records URL"} string nextRecordsUrl)
-                                                returns @display {label: "SOQL Result"} SoqlResult|Error {
-        json res = check self.get(nextRecordsUrl);
-        return toSoqlResult(res);
+    # Notifies Salesforce servers that the upload of job data is complete
+    # + bulkJobId - Id of the bulk job
+    # + return - error?
+    isolated remote function closeJob(string bulkJobId) returns error|future<BulkJobInfo|error> {
+        final string path = utils:prepareUrl([API_BASE_PATH, JOBS, INGEST, bulkJobId]);
+        record{} payload = {"state" : "UploadComplete"};
+        http:Response response = check self.salesforceClient->patch(path, payload);
+        if response.statusCode != 200 {
+            return error("Error occurred while closing the bulk job. " + 
+                (check response.getJsonPayload()).toString());
+        }
+        worker A returns BulkJobInfo|error {
+            while true {
+                runtime:sleep(2);
+                http:Response jobStatus = check self.salesforceClient->get(path);
+                if jobStatus.statusCode != 200 {
+                    return error("Error occurred while checking the status of the bulk job. " + 
+                        (check jobStatus.getJsonPayload()).toString());
+                } else {
+                    json responsePayload = check jobStatus.getJsonPayload();
+                    BulkJobInfo jobInfo = check responsePayload.cloneWithType(BulkJobInfo);
+                    if jobInfo.state == "JobComplete" || jobInfo.state == "Failed" || jobInfo.state == "Aborted" {
```

Comment:
shall we define these as enum and use them here

Reply:
Updated in [update code according to comments](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/311/commits/cd6ae536defa51bd247b84742f1c480c2cf80ab1)